More natural syntax.

beregond opened the pull request:
What was done:
- `ensure` alias
- `to_be_*` and `is_*` aliases (except for `Not` assertion, because of name clash with `IsNot`)
- aliases for numeric assertions like `GreaterOrEqual`, since this way is much more simple for codebase (no weird `_or_` infixes handling)
- special methods `does` for `Predicate` and `does_not` for `Not` (also because of above reason)

Closes #7.

beregond committed:
Added ensure alias.

beregond committed:
Added aliases and prefixes for assertions.

beregond committed:
Refactor of is_assertion.

beregond committed:
Added aliases for numeric assertions.

beregond committed:
Updated docs.

beregond committed:
Refactor and coverage.

beregond committed:
Updated authors.

dgilland commented on verify/runners.py:
```diff
@@ -108,3 +110,58 @@ def __call__(self, *assertions):
+
+
+def _reserved_names(name):
+    if name == 'does':
```
I wonder if a better name for these might be `to_pass`/`passes` and `to_fail`/`fails` for `Predicate` and `Not` respectively?

dgilland commented on tests/test_verify.py:
```diff
@@ -92,3 +141,14 @@ def test_assert_raises(meth, value, arg):
     assert opts['msg'] in str(exc.value)
+
+
+def test_expect_and_ensure_are_aliases():
```
`test_expect_and_ensure_are_aliases` and `test_assertions_aliases` could be a single, parametrized test like:

``` python
@pytest.mark.parametrize('obj,alias', [
    (expect, ensure),
    (Greater, GreaterThan),
    ...snip...
])
def test_aliases(obj, alias):
    assert obj is alias
```

dgilland commented on README.rst:
```diff
@@ -107,6 +107,57 @@ And if you'd prefer to see ``assert`` being used, all ``verify`` assertions will
+Reserved names
+--------------
+
+Things that doesn't work as expected:
```
Should be `Things that don't...` instead of `doesn't`.

dgilland commented on README.rst:
```diff
@@ -107,6 +107,57 @@ And if you'd prefer to see ``assert`` being used, all ``verify`` assertions will
+    # Both above lines are the same as:
+    expect(some_var).Int().LessOrEqual(5).NotList()
+
+All assertions can be also used as lowercased methods:
```
Instead of `lowercased methods`, either `underscore case` or `snake case` (https://en.wikipedia.org/wiki/Snake_case).

dgilland commented on README.rst:
```diff
@@ -107,6 +107,57 @@ And if you'd prefer to see ``assert`` being used, all ``verify`` assertions will
+.. code-block:: python
+
+    expect(value).is_not(predicate)  # translated into IsNot assertion
+    expect('v').in('verify')  # syntax error, try `to_be_in`
```
Should `expect(value).is(...)` be included here? Or maybe we can handle `Is` and `In` as `is_` and `in_` respectively?

dgilland commented on README.rst:
```diff
@@ -116,83 +167,83 @@ All of the validators in ``verify`` are callables that can be used in two contex
-``Decreasing``          Assert that ``a`` is monotonically decreasing.
-``StrictlyDecreasing``  Assert that ``a`` is strictly decreasing.
-======================  ===========
+The available validators           are:
```
Looks like some extra spacing between "validators" and "are".

dgilland commented:
Just a few minor comments, but everything looks great overall!

Thanks for this! :+1:

beregond committed:
Fixed typos.

beregond committed:
Fixed special cases.

beregond commented:
@dgilland 
- Fixed typos
- Added `is` as special case in docs
- Added `is_` and `in_` aliases

About `does` and `does_not` aliases: I've made them, because it is easier for me to build sentence with these words (`pass` and `fail` are a bit harder) yet I think they could be very hand. Anyway there are three options:
- Keep only `does` and `does_not`
- Keep only `passes`/`to_pass` and `fails`/`to_fail`
- Add both above to the codebase

Please make this decision, it's your call :)

dgilland commented:
Let's go ahead and keep `does/does_not` and add `passes/to_pass` and `fails/to_fail`.

beregond committed:
Extended special cases for assertions names.

beregond commented:
@dgilland I think it's all ready now

dgilland commented:
Awesome! Thanks!
